src/ffi/android.rs

```diff
@@ -201,19 +222,28 @@ pub unsafe extern "system" fn Java_org_c4dt_artiwrapper_TorLibApi_torRequest(
     headers_j: JObject,
     body_j: jbyteArray,
 ) -> jobject {
-    let (cache_dir, request) =
-        match make_request(env, cache_dir_j, method_j, url_j, headers_j, body_j) {
-            Ok(v) => v,
-            Err(e) => {
-                let _ = env.throw((
-                    "org/c4dt/artiwrapper/TorLibException",
-                    format!("make request: {:?}", e),
-                ));
-                return JObject::null().into_inner();
-            }
-        };
+    let request = match make_request(env, method_j, url_j, headers_j, body_j) {
+        Ok(v) => v,
+        Err(e) => {
+            let _ = env.throw((
+                "org/c4dt/artiwrapper/TorLibException",
+                format!("make request: {:?}", e),
+            ));
+            return JObject::null().into_inner();
+        }
+    };
     trace!("Request: {:?}", request);
 
+    let cache_dir = match get_cache_dir(env, cache_dir_j) {
+        Ok(v) => v,
+        Err(e) => {
+            let _ = env.throw((
+                "org/c4dt/artiwrapper/TorLibException",
+                format!("get cache dir: {:?}", e),
+            ));
+            return JObject::null().into_inner();
+        }
```

Comment:
Also a good case of `unwrap_or_else`.

While I'm at it, you can even extract it into another function, as to avoid duplicating `"org/c4dt/artiwrapper/TorLibException"` in other parts of the code.